src/azure-cli-core/azure/cli/core/extension/_resolve.py

```diff
@@ -71,7 +74,15 @@ def resolve_from_index(extension_name, cur_version=None, index_url=None):
     candidates_sorted = sorted(candidates, key=lambda c: parse_version(c['metadata']['version']), reverse=True)
     logger.debug("Candidates %s", [c['filename'] for c in candidates_sorted])
     logger.debug("Choosing the latest of the remaining candidates.")
-    chosen = candidates_sorted[0]
+
+    if cur_version:
+        try:
+            chosen = [c for c in candidates_sorted if c['metadata']['version'] == cur_version][0]
+        except Exception:
```

Comment:
Is it possible to catch specific exception here?

Reply:
catch IndexError instead.